### PR TITLE
Add `GetBoundingRectAfterMutations` to EmbeddedViewParams to calculate the final bounding rect for platform view

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

Adds a method to calculate the final bounding rect of the platform view.
Clippings are currently ignored

## Related Issues

https://github.com/flutter/flutter/issues/59821

## Tests

embedded_view_params_unittests.cc

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: flow/embedded_views.h [hunk 1]
```diff
@@ -201,6 +201,33 @@ class EmbeddedViewParams {
            sizePoints == other.sizePoints &&
            mutatorsStack == other.mutatorsStack;
   }
+
+  // Clippings are ignored in this operation
+  // TODO(cyanglaz): Factoring clipping.
+  // https://github.com/flutter/flutter/issues/59821
+  SkRect GetBoundingRectAfterMutations() const {
+    SkPath path;
+    SkRect starting_rect = SkRect::MakeSize(sizePoints);
```
Comment (blasten) [inline flow/embedded_views.h:210]: does it still need the `offsetPixels`?

Comment (cyanglaz) [reply flow/embedded_views.h:210]: I think mutator stack should contain all the offset informations already, By querying through the stack, we have the final offset

**Comment by cyanglaz** [comment]
@blasten updated with our offline discussion

File: flow/embedded_views.h [hunk 2]
```diff
@@ -186,21 +186,48 @@ class EmbeddedViewParams {
...
+    final_bounding_rect_ = other.final_bounding_rect_;
   };
 
-  SkPoint offsetPixels;
-  SkSize sizePoints;
-  MutatorsStack mutatorsStack;
+  const SkPoint& offsetPixels() const { return offset_pixels_; };
+  const SkSize& sizePoints() const { return size_points_; };
+  const MutatorsStack& mutatorsStack() const { return mutators_stack_; };
+  const SkRect& finalBoundingRect() const { return final_bounding_rect_; }
```
Comment (blasten) [inline flow/embedded_views.h:215]: nit: could we document these methods?

File: shell/platform/embedder/embedder_layers.cc [hunk 3]
```diff
@@ -180,10 +180,10 @@ void EmbedderLayers::PushPlatformViewLayer(
   layer.platform_view = platform_views_referenced_.back().get();
 
   const auto layer_bounds =
-      SkRect::MakeXYWH(params.offsetPixels.x(),                          //
-                       params.offsetPixels.y(),                          //
-                       params.sizePoints.width() * device_pixel_ratio_,  //
-                       params.sizePoints.height() * device_pixel_ratio_  //
+      SkRect::MakeXYWH(params.offsetPixels().x(),                          //
```
Comment (blasten) [inline shell/platform/embedder/embedder_layers.cc:183]: is this just `finalBoundingRect()` now?

Comment (cyanglaz) [reply shell/platform/embedder/embedder_layers.cc:183]: I think the size is not the same as the bounding rect. The size points are the original size before applying mutation.

Comment (cyanglaz) [reply shell/platform/embedder/embedder_layers.cc:183]: @chinmaygarde I thought params.sizePoints did not consider scaling and rotation at all? Does embedder free from scaling and rotation layers?

File: flow/embedded_views.h (hunk 2 above)
Comment (cyanglaz) [reply flow/embedded_views.h:215]: done

File: shell/platform/embedder/embedder_layers.cc (hunk 3 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder_layers.cc]: I believe the embedder is supposed to apply the scaling and rotation themselves from the data present in the mutator stack to get the final width and height. I believe there is a test for this in the embedder unit-tests harness and this should be documented in the header.

Comment (cyanglaz) [reply shell/platform/embedder/embedder_layers.cc]: I meant this rect returned from this method does not include rotation and scaling. Want to make sure if the original size used here is intended.

Comment (chinmaygarde) [reply shell/platform/embedder/embedder_layers.cc]: Yes. It is.

**Pull request merged**